internal/proxy/meta_cache.go

```diff
@@ -304,7 +304,14 @@ func (m *MetaCache) GetCollectionInfo(ctx context.Context, database, collectionN
 	if !ok || !collInfo.isCollectionCached() {
 		tr := timerecord.NewTimeRecorder("UpdateCache")
 		metrics.ProxyCacheStatsCounter.WithLabelValues(fmt.Sprint(paramtable.GetNodeID()), method, metrics.CacheMissLabel).Inc()
-		coll, err := m.describeCollection(ctx, database, collectionName, 0)
+		var coll *milvuspb.DescribeCollectionResponse
+		var err error
+		if collectionID != 0 {
+			coll, err = m.describeCollection(ctx, database, "", collectionID)
+		} else {
+			coll, err = m.describeCollection(ctx, database, collectionName, 0)
```

Comment:
Divide this method to two versions, one looks up by collection name, and one looks up by collection ID